### PR TITLE
Handle corrupted suite registration files better.

### LINKED ISSUE #1589: Handle corrupted suite registration files.
If a suite registration file under `$HOME/.cylc/REGDB/` is corrupted (e.g. empty, instead of containing suite title and path) it can cause some things to fail. e.g.:
 * `cylc db print` will abort when it tries to print that suite - should just issue a warning and continue
 * `cylc scan` will fail badly if any old suite is running (in which case scan has to load passphrases for all registered suites).
```
$ cylc scan
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/lib64/python2.7/threading.py", line 811, in __bootstrap_inner
    self.run()
  File "/usr/lib64/python2.7/threading.py", line 764, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib64/python2.7/multiprocessing/pool.py", line 376, in _handle_results
    task = get()
TypeError: ('__init__() takes exactly 2 arguments (1 given)', <class 'cylc.registration.RegistrationError'>, ())
```

### PR DESCRIPTION
**Pull request by hjoliver**
Close #1589.

### CODE REVIEW

File: bin/cylc-restart [hunk 1]
```diff
@@ -479,4 +480,9 @@ class restart(scheduler):
 
 
 if __name__ == '__main__':
-    main("restart", restart)
+    try:
+        main("restart", restart)
+    except Exception as exc:
+        if cylc.flags.debug:
+            raise
+        sys.exit(exc)
```
Comment (hjoliver) [inline bin/cylc-restart]: Made consistent with the other commands, which only dump a traceback if ```--debug``` is used.

Comment (hjoliver) [inline bin/cylc-restart]: (Made consistent with other commands)

**Comment by hjoliver** [comment]
@matthewrmshin - please review.  Small bug fix.

**Comment by matthewrmshin** [comment]
@benfitzpatrick please sanity check.

**Comment by benfitzpatrick** [comment]
Looks fine to me.

**Pull request merged**